--- jdk12u-jdk-12.0.2-ga/make/autoconf/platform.m4.orig	2019-07-16 18:38:28.000000000 +0200
+++ jdk12u-jdk-12.0.2-ga/make/autoconf/platform.m4	2022-02-10 23:23:27.419607985 +0100
@@ -30,17 +30,19 @@
 [
   # First argument is the cpu name from the trip/quad
   case "$1" in
-    x86_64*x32)
-      VAR_CPU=x32
-      VAR_CPU_ARCH=x86
-      VAR_CPU_BITS=32
-      VAR_CPU_ENDIAN=little
-      ;;
     x86_64)
       VAR_CPU=x86_64
       VAR_CPU_ARCH=x86
       VAR_CPU_BITS=64
       VAR_CPU_ENDIAN=little
+      case "$host" in
+        *x32)
+          VAR_CPU=x32
+          VAR_CPU_ARCH=x86
+          VAR_CPU_BITS=32
+          VAR_CPU_ENDIAN=little
+          ;;
+      esac
       ;;
     i?86)
       VAR_CPU=x86
--- jdk21u-jdk-21-ga/./src/java.base/share/classes/jdk/internal/util/Architecture.java.orig	2023-08-09 22:19:41.000000000 +0200
+++ jdk21u-jdk-21-ga/./src/java.base/share/classes/jdk/internal/util/Architecture.java	2023-09-21 13:14:27.515908810 +0200
@@ -36,6 +36,7 @@
     OTHER,      // An unknown architecture not specifically named
     X64,        // Represents AMD64 and X86_64
     X86,
+    X32,
     AARCH64,
     ARM,
     RISCV64,
